testplan/web_ui/testing/src/Report/BatchReport.js

```diff
@@ -117,54 +127,87 @@ class BatchReport extends React.Component {
+        axios
+          .get(`/api/v1/reports/${uid}`)
+          .then((response) => {
+            const rawReport = response.data;
+            if (rawReport.version === 2) {
+              const assertionsReq = axios.get(
+                `/api/v1/reports/${uid}/attachments/` +
+                `${rawReport.assertions_file}`,
+                { transformResponse: parseToJson }
+              );
+              const structureReq = axios.get(
+                `/api/v1/reports/${uid}/attachments/` +
+                `${rawReport.structure_file}`,
+                { transformResponse: parseToJson }
+              );
+              axios
+                .all([assertionsReq, structureReq])
+                .then(
+                  axios.spread((assertionsRes, structureRes) => {
+                    if (!assertionsRes.data) {
+                      console.error(assertionsRes);
+                      alert(
+                        "Failed to parse assertion datails!\n" +
+                        "Please report this issue to the Testplan team."
+                      );
+                    }
+                    const mergedReport = MergeSplittedReport(
+                      rawReport,
+                      assertionsRes.data,
+                      structureRes.data
                     );
-                  }
-                  const mergedReport = MergeSplittedReport(
-                    rawReport,
-                    assertionsRes.data,
-                    structureRes.data
-                  );
-                  this.setReport(this.updateReportUID(mergedReport, uid));
-                })
-              ).catch(this.setError);
-          } else {
-            this.setReport(this.updateReportUID(rawReport, uid));
-          }
-        })
-        .catch(this.setError);
+                    this.setReport(this.updateReportUID(mergedReport, uid));
+                  })
+                ).catch(this.setError);
+            } else {
+              this.setReport(this.updateReportUID(rawReport, uid));
+            }
+          })
+          .catch(this.setError);
+        break;
     }
   }
 
   updateReportUID(report, uid) {
     return { ...report, uid };
   }
 
+  /**
+    * Auto-select the first failed entry in the report when it is first loaded.
+    * @param {reportNode} reportEntry - the current report entry to select from.
+    * @return {Array[string]} List of UIDs of the currently selected entries.
+    */
+  findFirstFailure(reportEntry) {
+    const selection = [reportEntry.uid];
+    if (reportEntry.category === "testcase"
+      || reportEntry.entries.length === 0) {
+      return selection;
+    } else {
+      for (let entry in reportEntry.entries) {
+        if (reportEntry.entries[entry].status === "failed"
+          || reportEntry.entries[entry].status === "error") {
+          return selection.concat(
+            this.findFirstFailure(reportEntry.entries[entry])
+          );
+        }
+      }
+    }
+  }
+
```

Comment:
I would move this to reportUtils.js. Technically, this works independent of the BatchReport as a routine.

Reply:
Agree, moving to reportUtils